### PR TITLE
SG-13264: TankQDialog does not work in Python 3

### PR DESCRIPTION
**Pull request by jfboismenu**
I simply had to regenerate the Qt resource file for TankQDialog. I also added a test for it.

Note that I found some weird race condition that seemed to only happen when running tests using `PySide2` and `pytest`. We would get this error:

`Internal c++ Object (Invoker) Already Deleted`, which is the object in the engine that is responsible for forwarding async and sync calls to the main thread. Moving Qt's initialization at the beginning solved this.

I also added the `qt` folder to the coverage. It brings us down a bit, but it's for a good cause. 😄

### CODE REVIEW

File: tests/platform_tests/test_engine.py [hunk 1]
```diff
@@ -141,33 +141,37 @@ def test_empty_environment(self):
         )
 
 
+@skip_if_pyside_missing
```
Comment (jfboismenu) [inline tests/platform_tests/test_engine.py]: Class decorators were introduced in Python 2.6, so we can use them here.

File: tests/run_tests.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
```
Comment (jfboismenu) [inline tests/run_tests.py]: Allows to run tests on Un*x without the shell wrapper

File: .coveragerc [hunk 3]
```diff
@@ -17,7 +17,6 @@ omit=
     # We can't unit test our uis, so we're not going to put them inside the coverage
     *tank/authentication/ui/*
     *tank/authentication/sso_saml2/*
-    *tank/platform/qt/*
```
Comment (willis102) [inline .coveragerc]: I'm not sure I understand why we want to stop omitting this directory?

Comment (jfboismenu) [reply .coveragerc]: Back in the day we believed we wouldn't/cloudn't unit test our GUIs. Well, here we are. 😄

Comment (willis102) [reply .coveragerc]: 👍 A little slow on the uptake today...

**Pull request merged**